src/components/v2/V2Project/LaunchProjectPayer/AdvancedOptionsCollapse.tsx

```diff
@@ -1,129 +1,142 @@
 import { t, Trans } from '@lingui/macro'
-import { Input, Space, Switch } from 'antd'
+import { Form, FormInstance, Input, Space, Switch } from 'antd'
 import TooltipLabel from 'components/TooltipLabel'
 import { V2ProjectContext } from 'contexts/v2/projectContext'
-import { Dispatch, SetStateAction, useContext, useState } from 'react'
+import { useContext, useState } from 'react'
 import * as constants from '@ethersproject/constants'
 import { EthAddressInput } from 'components/inputs/EthAddressInput'
 import { MinimalCollapse } from 'components/MinimalCollapse'
+import { FormImageUploader } from 'components/inputs/FormImageUploader'
+
+import { AdvancedOptionsFields } from './LaunchProjectPayerModal'
+
+const defaultAdvancedOptions: AdvancedOptionsFields = {
+  memo: '',
+  memoImageUrl: undefined,
+  tokenMintingEnabled: true,
+  preferClaimed: false,
+  customBeneficiaryAddress: undefined,
+}
 
 export default function AdvancedOptionsCollapse({
-  memo,
-  setMemo,
-  customBeneficiaryAddress,
-  setCustomBeneficiaryAddress,
-  tokenMintingEnabled,
-  setTokenMintingEnabled,
-  preferClaimed,
-  setPreferClaimed,
+  form,
 }: {
-  memo: string | undefined
-  setMemo: Dispatch<SetStateAction<string | undefined>>
-  customBeneficiaryAddress: string | undefined
-  setCustomBeneficiaryAddress: Dispatch<SetStateAction<string | undefined>>
-  tokenMintingEnabled: boolean
-  setTokenMintingEnabled: Dispatch<SetStateAction<boolean>>
-  preferClaimed: boolean
-  setPreferClaimed: Dispatch<SetStateAction<boolean>>
+  form: FormInstance<AdvancedOptionsFields>
 }) {
   const { tokenAddress } = useContext(V2ProjectContext)
 
+  // need state for this field to update dom
+  const [tokenMintingEnabled, setTokenMintingEnabled] = useState<boolean>(
+    form.getFieldValue('tokenMintingEnabled') === false ? false : true,
+  )
+
   const [customBeneficiaryEnabled, setCustomBeneficiaryEnabled] =
-    useState<boolean>(false)
+    useState<boolean>(Boolean(form.getFieldValue('customBeneficiaryAddress')))
```

Comment:
**suggestion** all this can be replaced with antd `useWatch`

Reply:
Do you mean like: 

`const customBeneficiaryEnabled = Boolean(useWatch('customBeneficiaryAddress', form))`      ?